crates/transaction-emitter-lib/src/transaction_generator/mod.rs

```diff
@@ -75,64 +76,82 @@ pub async fn create_txn_generator_creator(
+    ) -> Box<dyn TransactionGeneratorCreator> {
+        if use_account_pool {
+            Box::new(AccountsPoolWrapperCreator::new(inner, accounts_pool))
+        } else {
+            inner
+        }
+    }
+
     for transaction_mix in transaction_mix_per_phase {
         let mut txn_generator_creator_mix: Vec<(Box<dyn TransactionGeneratorCreator>, usize)> =
             Vec::new();
         for (transaction_type, weight) in transaction_mix {
-            let txn_generator_creator: Box<dyn TransactionGeneratorCreator> = match transaction_type
-            {
-                TransactionType::CoinTransfer {
-                    invalid_transaction_ratio,
-                } => Box::new(P2PTransactionGeneratorCreator::new(
-                    txn_factory.clone(),
-                    SEND_AMOUNT,
-                    all_addresses.clone(),
-                    *invalid_transaction_ratio,
-                )),
-                TransactionType::AccountGeneration {
-                    add_created_accounts_to_pool,
-                    max_account_working_set,
-                    creation_balance,
-                } => Box::new(AccountGeneratorCreator::new(
-                    txn_factory.clone(),
-                    all_addresses.clone(),
-                    accounts_pool.clone(),
-                    *add_created_accounts_to_pool,
-                    *max_account_working_set,
-                    *creation_balance,
-                )),
-                TransactionType::NftMintAndTransfer => Box::new(
-                    NFTMintAndTransferGeneratorCreator::new(
-                        txn_factory.clone(),
-                        all_accounts.get_mut(0).unwrap(),
-                        txn_executor,
-                        num_workers,
-                    )
-                    .await,
-                ),
-                TransactionType::PublishPackage => {
-                    Box::new(PublishPackageCreator::new(txn_factory.clone()))
-                },
-                TransactionType::CallCustomModules {
-                    entry_point,
-                    num_modules,
-                    use_account_pool,
-                } => Box::new(
-                    CallCustomModulesCreator::new(
+            let txn_generator_creator: Box<dyn TransactionGeneratorCreator> =
+                match transaction_type {
+                    TransactionType::CoinTransfer {
+                        invalid_transaction_ratio,
+                        sender_use_account_pool,
+                    } => wrap_accounts_pool(
+                        Box::new(P2PTransactionGeneratorCreator::new(
+                            txn_factory.clone(),
+                            SEND_AMOUNT,
+                            all_addresses.clone(),
+                            *invalid_transaction_ratio,
+                        )),
+                        *sender_use_account_pool,
+                        accounts_pool.clone(),
```

Comment:
Where do you generate this account pool? I don't see corresponding change? Also, how many accounts do we generate?

Reply:
TransactionType::AccountGeneration has add_created_accounts_to_pool field. 
so in order to use this, you need to have two phases, one with generation and second with your transaction